pylops/basicoperators/firstderivative.py

```diff
@@ -100,34 +100,39 @@ def __init__(
         self.kind = kind
         self.edge = edge
         self.order = order
-        self._register_multiplications(self.kind, self.order)
 
-    def _register_multiplications(
```

Comment:
The reason we add this is because you are better of finding out once upfront which method to use instead of going through if...else statements every time the operator forward and adjoint methods are called... I have reinstated this :)

Reply:
Ok understood, yeah it's better instead of putting the if...else statements in `_matvec` and `_rmatvec` and getting it called again and again